home_assistant_streamdeck_yaml.py

```diff
@@ -62,9 +64,281 @@
+                self._max = float(data["attributes"]["max"])
+                self._step = float(data["attributes"]["step"])
+            except KeyError as e:
+                console.log(f"KeyError raised when trying to  set attributes {e}")
+        else:
+            try:
+                self._min = float(self.attributes["min"])
+                self._max = float(self.attributes["max"])
+                self._step = float(self.attributes["step"])
+            except KeyError as e:
+                console.log(f"KeyError raised when trying to set update_attributes")
+        
+        if self.state_attribute is None:
+            self._state = float(data["state"])
+        else:
+            try:
+                if data["attributes"][self.state_attribute] is None:
+                    self._state = 0
+                else:
+                    self._state = float(data["attributes"][self.state_attribute])
+            except KeyError as e:
+                console.log(f"Could not find attribute {self.state_attribute}")
+                self._state = 0
+
+    def get_attributes(self) -> dict[str, int]:
+        """Returns all home assistant entity attributes"""
+        return {
+            "min": self._min,
+            "max": self._max,
+            "step": self._step,
+            "state": self._state,
+        }
+        
+    def increment_state(self, value: float):
+        """Increments the value of the dial with checks for the minimal and maximal value"""
+        num: float = self._state + value * self._step
+        num = min(self._max, num)
+        num = max(self._min, num)
+        self._state = num
+    
+    def set_state(self, value: float):
+        """Sets the value of the dial without checks for the minimal and maximal value"""
+        self._state = value
+    
+    
+    @classmethod
+    def templatable(cls: type[Dial]) -> set[str]:
+        """Return if an attribute is templatable, which is if the type-annotation is str."""
+        schema = cls.schema()
+        properties = schema["properties"]
+        return {k for k, v in properties.items() if v["allow_template"]}
+    
+    def rendered_template_dial(
+        self,
+        complete_state: StateDict,
+    ) -> Dial:
+        dct = self.dict(exclude_unset=True)
+        for key in self.templatable():
+            if key not in dct:
+                continue
+            val = dct[key]
+            if isinstance(val, dict):
+                for k, v in val.items():
+                    val[k] = _render_jinja(v, complete_state, self)
+            else:
+                dct[key] = _render_jinja(val, complete_state, self)
+        return Dial(**dct)
+    
+    #LCD/Touchscreen management
+    def render_lcd_image(
+        self,
+        complete_state: StateDict,
+        key: int, #Key needs to be from sorted dials
+        size: tuple[int,int],
+        icon_mdi_margin: int = 0,
+        font_filename: str = DEFAULT_FONT,
+    ) -> Image.Image:
+        try:
```

Comment:
why is this big try/catch block needed? You should try to keep try/catch blocks as small as possible, only wrapping the code which raises the exceptions you want to catch.